service/src/main.rs

```diff
@@ -820,19 +810,18 @@ fn subscribe_to_parentchain_new_headers<E: EnclaveBase + Sidechain>(
 	parentchain_handler: Arc<ParentchainHandler<ParentchainApi, E>>,
 	mut last_synced_header: Header,
 ) -> Result<(), Error> {
-	let (sender, receiver) = channel();
-	//TODO: this should be implemented by parentchain_handler directly, and not via
-	// exposed parentchain_api. Blocked by https://github.com/scs/substrate-api-client/issues/267.
-	parentchain_handler
+	// TODO: this should be implemented by parentchain_handler directly, and not via
+	// exposed parentchain_api
+	let mut subscription = parentchain_handler
 		.parentchain_api()
-		.subscribe_finalized_heads(sender)
+		.subscribe_finalized_heads()
 		.map_err(Error::ApiClient)?;
 
 	loop {
-		let new_header: Header = match receiver.recv() {
-			Ok(header_str) => serde_json::from_str(&header_str).map_err(Error::Serialization),
-			Err(e) => Err(Error::ApiSubscriptionDisconnected(e)),
-		}?;
+		let new_header = subscription
+			.next()
+			.ok_or(Error::ApiSubscriptionDisconnected)?
+			.map_err(|e| Error::ApiClient(e.into()))?;
```

Comment:
Ehm, this does not look reasonable, first we create an error from a None ourselves, which we map afterwards to create yet another error?

Am I misunderstanding something?

Reply:
Yes, you're overlooking the question mark after the `ok_or`.

Reply:
hahahha, yes.